srcjs/shinyapp.js

```diff
@@ -631,6 +631,38 @@ var ShinyApp = function() {
     window.location.reload();
   });
 
+  addMessageHandler('shiny-insert-ui', function (message) {
+      var targets = $(message.selector);
+      if (targets.length === 0) {
+        // render the HTML and deps to a null target, so
+        // the side-effect of rendering the deps, singletons,
+        // and <head> still occur
+        exports.renderHtml($([]), message.content.html, message.content.deps);
+      } else {
+        targets.each(function (i, target) {
+          var container = document.createElement(message.container);
+          insertAdjacentElement(message.where, target, container);
+          exports.renderContent(container, message.content);
+          $(container).trigger('shown');
+          return message.multiple;
+        });
+      }
+    });
+
+  addMessageHandler('shiny-remove-ui', function (message) {
+    var els = $(message.selector);
+    els.each(function (i, el) {
+      exports.unbindAll(el, true);
+      $(el).trigger('hide');
```

Comment:
How about this, then? I added a call to `sendImageSize` from inside the `unbindOutputs` function. This way the function mimics `bindOutputs` in that it calls both `sendImageSize` and `sendOuputHiddenState`. I realize this may not always be necessary for `unbindOutputs`, but in practice, we pretty much only unbind things when we're modifying them or removing them, so, most times, this will be necessary. And for the other times, it doesn't hurt, right? It feels like this is the cleaner solution... But if any of you have objections, here are two alternatives:

1- Change `shiny-remove-ui` back to this:
```js
addMessageHandler('shiny-remove-ui',
    function(message) {
      let els = $(message.selector);
      els.each((i, el) => {
        exports.unbindAll(el, true);
        $(el).trigger('hide');
        $(el).hide();
        $(el).trigger('hidden');
        $(el).remove();
        // If `multiple` is false, returning false terminates the function
        // and no other elements are removed; if `multiple` is true,
        // returning true continues removing all remaining elements.
        return message.multiple;
      });
    }
  );
```


2- Call `sendImageSize` directly from `shiny-remove-ui` (no need to worry about `sendOutputHiddenState` because, as Joe mentions, that is taken care of by the call to `unbindAll`):
```js
addMessageHandler('shiny-remove-ui',
    function(message) {
      let els = $(message.selector);
      els.each((i, el) => {
        exports.unbindAll(el, true);
        setTimeout(sendImageSize, 0);
        $(el).remove();
        // If `multiple` is false, returning false terminates the function
        // and no other elements are removed; if `multiple` is true,
        // returning true continues removing all remaining elements.
        return message.multiple;
      });
    }
  );
```